assembler.ts

```diff
@@ -831,6 +826,11 @@ function resolveLabelsPass(
             value = undefined;
           } else {
             value = v.reg;
+
+            const method = ops[instr.op];
+            if(value > 0 && (typeof method.out === "number" ? method.out === vi : method.out?.includes(vi))) {
```

Comment:
There might not be any parameter names, so we also need something like "markRead" for non-output params.
Also, for "call" we should treat all params as output.

Reply:
Ah yes. About call, always flagging the params as output is technically not correct. I could implement it correctly but this would require the parameter flagging to happen in another pass after main & the subs have all been assembled I believe.

Reply:
That would be more correct, but like you said in the initial comment, the game seems to deal with it just fine if we have `true` instead of `false`. 
I don't know if it's worth implementing multiple passes when the game doesn't seem to care. Although it would probably be helpful to have correct info for this if we want to improve the register allocator. So either way seems fine to me.

Reply:
I implemented a call graph & am propagating the write flag without an additional pass. Probably unnecessary, but if a call graph might be useful anyway...